### PR TITLE
Change how dataset to validate against is specified for `filenameExists` rule

### PR DESCRIPTION
**Pull request by GiaJordan**
Specify the dataset with files that should be validated against as a parameter in the cli and api instead of in the data model

### CODE REVIEW

File: schematic/models/commands.py [hunk 1]
```diff
@@ -150,6 +155,7 @@ def submit_manifest(
     hide_blanks,
     restrict_rules,
     project_scope,
+    dataset_scope,
```
Comment (BryanFauble) [inline schematic/models/commands.py:158]: nit: Could add in some argument types to the function

File: schematic/models/validate_attribute.py [hunk 2]
```diff
@@ -2027,13 +2028,18 @@ def filename_validation(
             errors: list[str] Error details for further storage.
             warnings: list[str] Warning details for further storage.
         """
+
+        if dataset_scope is None:
```
Comment (BryanFauble) [inline schematic/models/validate_attribute.py:2032]: If this cannot be none I would expect that the argument of the function is not optional (ie: No `= None`).

Could you also add this to the Args of the docstring?

Comment (GiaJordan) [reply schematic/models/validate_attribute.py:2033]: @BryanFauble yeah that makes sense thanks for catching

**Comment by GiaJordan** [comment]
At the time of writing the tests have passed, comments have been addressed, and this PR can be re-reviewed. If that happens while I'm out then one of the reviewers can merge to `develop`

**Review by BryanFauble — APPROVED** [review]
LGTM!

**Pull request merged**